components/sensor/tx20.rst

```diff
@@ -0,0 +1,78 @@
+Tx20/Tx23 Wind Speed/Direction Sensor
+=====================================
+
+.. seo::
+    :description: Instructions for setting up Tx20/Tx23 wind speed and direction sensors
+    :image: images/tx20.jpg
+    :keywords: Tx20
+
+The ``tx20`` sensor platform allows you to use your Tx20/Tx23
+(`amazon <https://www.amazon.de/Technoline-Tx-20/dp/B01HXZ3KLA>`__,
+`lacrosse_tx23 <https://www.lacrossetechnology.com/tx23-wind-sensor>`__) 
+wind direction and speed sensors with esphomelib. 
+The only required pin is the black or brown cable (which is used to transmit data.)
+The red cable should be connected to VCC (3.3V), green initiates the data transfer when connected to GND
+and should therefore be directly connected to GND and the yellow cable should be always connected to ground.
+The difference from the Tx20 to the Tx23 seems to be that the green cable does not matter anymore and the
+wind sensor is just sending a datagram every 2 seconds - pretty much the same as when the green cable was
+connected to GND in the Tx20.
+
+.. figure:: images/tx20.jpg
+    :align: center
+    :width: 50.0%
+
+    Tx20/Tx23 Wind Speed / Direction.
+
+.. _Amazon: https://www.amazon.de/Technoline-Tx-20/dp/B01HXZ3KLA
+.. _lacrosse_tx23: https://www.lacrossetechnology.com/tx23-wind-sensor
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+      - platform: tx20
+        wind_speed:
+          name: "Windspeed"
+        wind_direction_degrees:
+          name: "Winddirection Degrees"
+        pin: 
+          number: GPIO04
+          mode: INPUT
+
+    text_sensor:
+      - platform: template
+        name: "Textual Representation Wind Direction"
+        lambda: |-
+          return {id(tx20_id).get_wind_cardinal_direction()};
```

Comment:
This example config will not work because `tx20_id` is not defined.

One of two things can be done:

 1. Remove text_sensor from example config and move that into a short `.. note::` below the config variables (preferred from my side)
 2. add `tx20_id` to the example config